### PR TITLE
Add min periods parameter to Window

### PR DESCRIPTION
**Pull request by emilyreff7**
We want to set min_periods=1 on rolling windows to avoid defaulting to min_periods equal to the window size n, thereby eliminating the necessity of NaNs in the first n-1 rows for trailing window operations.

### CODE REVIEW

**Comment by cpcloud** [comment]
@emilyreff7 It looks like you have a lint error: https://circleci.com/gh/ibis-project/ibis/7540?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I'm going to looking into adding [`pep8speaks`](https://github.com/OrkoHunter/pep8speaks) to ibis so that these kinds of failures are immediately surfaced rather than having to hunt through CI.

You can also see these failures on the command line by running `make lint`.

**Comment by cpcloud** [comment]
@emilyreff7 Another thing that will prevent lint issues in a PR is to setup the commit hooks using `pre-commit`. Once you've created your environment you can run `pip install pre-commit` and then inside your ibis clone run `pre-commit install`. Then when you run `git commit` the lint and style checks will run and will prevent committing unless they pass.

File: ibis/pandas/tests/test_udf.py [hunk 1]
```diff
@@ -264,7 +264,29 @@ def my_mean(series):
...
+        .mean()
+        .reset_index(level=0, drop=True)
+    )
+    tm.assert_frame_equal(result, expected)
+
+
+def test_udaf_window_nan():
+    df = pd.DataFrame(
+        {
+            'a': np.arange(10, dtype=float).tolist(),
```
Comment (cpcloud) [inline ibis/pandas/tests/test_udf.py]: No need to call `tolist` here.

File: ibis/pandas/aggcontext.py [hunk 2]
```diff
@@ -421,7 +421,7 @@ def __init__(self, preceding, *args, **kwargs):
             )
             else 'both'
         )
-        super().__init__('rolling', preceding, *args, closed=closed, **kwargs)
+        super().__init__('rolling', preceding, *args, closed=closed, min_periods=1, **kwargs)
```
Comment (cpcloud) [inline ibis/pandas/aggcontext.py]: Should we add `min_periods=1` to `Cumulative` as well?

Comment (cpcloud) [reply ibis/pandas/aggcontext.py]: We could do this in a follow-up as well.

Comment (icexelloss) [reply ibis/pandas/aggcontext.py]: Just curious why does Cumulative needs min_period?

Comment (icexelloss) [reply ibis/pandas/aggcontext.py]: I see, it's one of the parameter sort of unique to expanding window in pandas:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.expanding.html

File: ibis/pandas/tests/test_udf.py (hunk 1 above)
Comment (emilyreff7) [reply ibis/pandas/tests/test_udf.py]: Removed in most recent commit

File: ibis/pandas/aggcontext.py (hunk 2 above)
Comment (emilyreff7) [reply ibis/pandas/aggcontext.py]: I will create a follow-up to add `min periods=1` to `Cumulative`

**Comment by emilyreff7** [comment]
> @emilyreff7 Another thing that will prevent lint issues in a PR is to setup the commit hooks using `pre-commit`. Once you've created your environment you can run `pip install pre-commit` and then inside your ibis clone run `pre-commit install`. Then when you run `git commit` the lint and style checks will run and will prevent committing unless they pass.

I ran the above but got an error once I got to the `git commit` step:

Errors: 
    Traceback (most recent call last):
      File "/home/emily/anaconda3/envs/ibis-dev/lib/python3.6/site-packages/virtualenv.py", line 2580, in <module>
        main()
      File "/home/emily/anaconda3/envs/ibis-dev/lib/python3.6/site-packages/virtualenv.py", line 831, in main
        symlink=options.symlink,
      File "/home/emily/anaconda3/envs/ibis-dev/lib/python3.6/site-packages/virtualenv.py", line 1123, in create_environment
        install_wheel(to_install, py_executable, search_dirs, download=download)
      File "/home/emily/anaconda3/envs/ibis-dev/lib/python3.6/site-packages/virtualenv.py", line 973, in install_wheel
        _install_wheel_with_search_dir(download, project_names, py_executable, search_dirs)
      File "/home/emily/anaconda3/envs/ibis-dev/lib/python3.6/site-packages/virtualenv.py", line 1060, in _install_wheel_with_search_dir
        call_subprocess(cmd, show_stdout=False, extra_env=env, stdin=script)
      File "/home/emily/anaconda3/envs/ibis-dev/lib/python3.6/site-packages/virtualenv.py", line 924, in call_subprocess
        raise OSError("Command {} failed with error code {}".format(cmd_desc, proc.returncode))
    OSError: Command /home/emily/.cache/p...hon3.6/bin/python3.6 - setuptools pip wheel failed with error code 2


I re-installed pre-commit using conda (`conta install -c conda-forge pre_commit`) and it resolved the issue.

Also now just waiting on the checks to run...have been running/queued up for 1+ hours now.

File: ibis/pandas/aggcontext.py (hunk 2 above)
Comment (emilyreff7) [reply ibis/pandas/aggcontext.py]: Actually do we need to even add `min_periods=1` to Cumulative? It looks like that is already the default behavior for Expanding windows.

Comment (cpcloud) [reply ibis/pandas/aggcontext.py]: Cool, then we don't need to do that. Thanks for looking into it.

**Comment by cpcloud** [comment]
@emilyreff7 Sometimes the checks take awhile if you haven't enabled the CI services on your fork. If you enable CircleCI and Azure Pipelines on your fork then the CI will run with a separate set of containers than the ones that are used to run CI for the `ibis-project` user. I can help you set that up if you want.

**Comment by cpcloud** [comment]
In any event, this LGTM.

**Pull request merged**